R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,214 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
+      gather(year, value, -GCAM_region_ID, -Non.CO2, -supplysector, -subsector, -stub.technology, -agg_sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L161.SSP15_EF
+    get_data(all_data, "temp-data-inject/L161.SSP34_EF")  %>%
+      gather(year, value, -GCAM_region_ID, -Non.CO2, -supplysector, -subsector, -stub.technology, -agg_sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L161.SSP34_EF
+    get_data(all_data, "temp-data-inject/L201.nonghg_steepness") ->
+      L201.nonghg_steepness
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+# This section takes in the non-CO2 emissions factors for SSP 1.5, 2, and 3.4 across sectors and
+# interpolates across years 2010-2100 in 5 year segments.
+
+    L161.SSP15_EF %>%
+      # Input data only contains 3 future years.  Need to include years between 2010-2100 in 5 year segments,
+      # and interpolate between each time segment.
+      tidyr::complete(year = SSP_FUTURE_YEARS,
+          tidyr::nesting(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector),
+          fill = list(value = NA)) %>%
+      mutate(year = as.integer(year))  %>%
+      group_by(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector) %>%
+      mutate(value = approx_fun(year, value))  %>%
+      # Now add regional information.
+      left_join_error_no_match(A_regions, by = c("GCAM_region_ID")) %>%
+      # Emission coefficients values are too long, so we round to the 10th decimal point.
+      mutate(emiss.coeff = round(value, 10)) %>%
+      ungroup() %>%
+      # Discard columns that are not needed.
+      select(-MAC_region, -bio_N2O_coef, -SO2_name, -GAINS_region, -GCAM_region_ID, -agg_sector, -value) ->
+    L251.ssp15_ef
```

Comment:
I would include a note here indicating what columns are in the resulting output (the `select` above tells us what's _not_ in, but that's not so useful).  If all of these tables are going to have the same set of columns, then you could just expand the comment at lines 64-65 to include this information and note that it applies to all of these tables.